R/predict.R

```diff
@@ -0,0 +1,104 @@
+#' Predict hazard, cumulative hazard or survival probability
+#'
+#' @param object An object of class \code{pam_xgb}
+#' @param newdata A data set containing the same covariates as used for model
+#' fitting. If of class \code{data.frame}, the function will transform the
+#' data to the PED format using the same settings as for the data used in
+#' estimation.
+#' @param type The type of prediction desired. Either hazard (\code{type = "hazard"}),
+#' cumulative hazard (\code{type = "cumu_hazard"}) or survival probability
+#' (\code{type = "surv_prob"}).
+#' @param ... Currently not used.
+#' @importFrom stats predict
+#' @return A matrix of predictions containing one row per
+#' observation (row in newdata) and 1 column per  specified time in the
+#' \code{times} argument.
+#' @seealso pamm
+#' @export
+predict.pamm <- function(
+  object,
+  newdata,
+  type = c("hazard", "cumu_hazard", "surv_prob"),
+  ...) {
+
+  type <- match.arg(type)
+
+  if (!is.ped(newdata)) {
+    newdata <- as_ped(object, newdata)
+  }
+
+  newdata[["pred"]] <- predict(unpam(object), newdata, type = "response")
```

Comment:
seems dangerous -- user supplied "pred" might be  overwritten here -- change to "..pred" throughout, maybe?

Reply:
nvrmind - we're not returning the entire `newdata` anyways